src/main/java/uk/gov/justice/digital/writer/curated/CuratedZoneCdcWriter.java

```diff
@@ -0,0 +1,37 @@
+package uk.gov.justice.digital.writer.curated;
+
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.SparkSession;
+import uk.gov.justice.digital.domain.model.SourceReference;
+import uk.gov.justice.digital.service.DataStorageService;
+import uk.gov.justice.digital.writer.Writer;
+
+import javax.inject.Inject;
+
+public class CuratedZoneCdcWriter extends Writer {
```

Comment:
If you add a `@Singleton` annotation to your writer classes micronaut can take care of wiring them in where used. See my earlier comment.

Reply:
I did consider using the @Singleton DI approach but wasn't sure what the performance impact would be given that the writer is invoked in a couple of places. Something definitely worth revisiting during the performance tuning.